### PR TITLE
Box Component

### PR DESCRIPTION
**Pull request by lucasaarcoverde**
#### What is the purpose of this pull request?

- Add Box Component.
- Add storybook documentation.

#### How should this be manually tested?

You can use the netlify build below. Or clone the repo and run storybook with yarn && yarn storybook

#### Screenshots or example usage

Captura de Tela 2020-07-27 às 17 13 28
Captura de Tela 2020-07-27 às 17 13 35

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [x] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: src/components/Box/index.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React, { forwardRef, Ref } from 'react'
+import { Box as WAIBox, BoxProps as WAIProps } from 'reakit'
+import { Box as TABox, BoxProps as TAProps } from 'theme-ui'
+
+type Props = WAIProps & TAProps
+function BaseBox({ as, ...props }: Props, ref: Ref<HTMLElement>) {
+  return <WAIBox {...props} ref={ref} as={as ?? TABox} />
+}
+
+export const Box = forwardRef(BaseBox)
```
Comment (matheusps) [inline src/components/Box/index.tsx:10]:
```suggestion
import React, { forwardRef, Ref } from 'react'
import { Box as A11yBox, BoxProps as A11yProps } from 'reakit'
import { Box as ThemeAwareBox, BoxProps as ThemeAwareProps } from 'theme-ui'

export type Props = A11yProps & ThemeAwareProps

function BaseBox({ as, ...props }: Props, ref: Ref<HTMLElement>) {
  return <WAIBox {...props} ref={ref} as={as ?? TABox} />
}

export const Box = forwardRef(BaseBox)
```

Comment (matheusps) [reply src/components/Box/index.tsx:10]: Just to keep the pattern

File: src/components/Box/Box.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+export function TheAsProp() {
+  return (
+    <Box>
+      <Box
+        px={3}
+        py={2}
+        sx={{ borderRadius: 10, maxWidth: 600, fontFamily: 'sans-serif' }}
+        bg="#F4F8FE"
+        color="#345EBD"
+      >
```
Comment (matheusps) [inline src/components/Box/Box.stories.tsx:35]: We need to add components to represent that in the future

**Comment by matheusps** [comment]
Missing CHANGELOG entry.

File: src/components/Box/index.tsx (hunk 1 above)
Comment (lucasaarcoverde) [reply src/components/Box/index.tsx]: done

**Pull request merged**